SETools GUI tests fail when PySide6 is installed

Hello,

When updating to SETools 4.6.0, Arch Linux users (who are building from source) are experiencing issues when the Python tests are run. I managed to reproduce the issue in a clean environment and it seems that:

- When PyQt6 is installed (through Arch Linux' package `python-pyqt6`) and not PySide6 (package `pyside6`), the testsuite in `tests/gui/` runs without any issue.
- When both PyQt6 and PySide6 are installed, the tests fail with many errors `TypeError: Need to pass a QWidget to addWidget!`

More precisely, both on release tag 4.6.0 and on current branch `main`, executing a test such as `test_bool_list` when PySide6 is installed (and a working X11 server is available through `$DISPLAY`) fails with:

```console
$ pytest tests -k test_bool_list
================================ test session starts ================================
platform linux -- Python 3.14.2, pytest-8.4.2, pluggy-1.6.0
PySide6 6.10.2 -- Qt runtime 6.10.2 -- Qt compiled 6.10.2
rootdir: /setools
configfile: pyproject.toml
plugins: qt-4.4.0
collected 1650 items / 1649 deselected / 1 selected                                 

tests/gui/widgets/criteria/test_boolean.py E                                  [100%]

====================================== ERRORS =======================================
_________________________ ERROR at setup of test_bool_list __________________________

mock_query = <Mock spec='PolicyQuery' id='139981473741328'>
request = <SubRequest 'list_widget' for <Function test_bool_list>>
qtbot = <pytestqt.qtbot.QtBot object at 0x7f4ff9d1f620>

    @pytest.fixture
    def list_widget(mock_query, request: pytest.FixtureRequest,
                    qtbot: QtBot) -> BooleanList:
        """Pytest fixture to set up the Boolean list widget."""
        marker = request.node.get_closest_marker("obj_args")
        kwargs = marker.kwargs if marker else {}
        w = BooleanList(request.node.name, mock_query, "name", **kwargs)
>       qtbot.addWidget(w)

tests/gui/widgets/criteria/test_boolean.py:17: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pytestqt.qtbot.QtBot object at 0x7f4ff9d1f620>
widget = <setoolsgui.widgets.criteria.boolean.BooleanList object at 0x7f5002a8c550>

    def addWidget(self, widget, *, before_close_func=None):
        """
        Adds a widget to be tracked by this bot. This is not required, but will ensure that the
        widget gets closed by the end of the test, so it is highly recommended.
    
        :param QWidget widget:
            Widget to keep track of.
    
        :kwparam before_close_func:
            A function that receives the widget as single parameter, which is called just before
            the ``.close()`` method gets called.
    
        .. note:: This method is also available as ``add_widget`` (pep-8 alias)
        """
        if not isinstance(widget, qt_api.QtWidgets.QWidget):
>           raise TypeError("Need to pass a QWidget to addWidget!")
E           TypeError: Need to pass a QWidget to addWidget!

/usr/lib/python3.14/site-packages/pytestqt/qtbot.py:188: TypeError
============================== short test summary info ==============================
ERROR tests/gui/widgets/criteria/test_boolean.py::test_bool_list - TypeError: Need to pass a QWidget to addWidget!
========================= 1649 deselected, 1 error in 0.74s =========================
```

(By the way, many tests fail, the test `test_bool_list` is the first one and others seem to fail in a similar way.) 

I am not familiar with Python Qt ecosystem and I am wondering what could cause `pytest` to use PySide6 instead of PyQt6 (it reports `PySide6 6.10.2 -- Qt runtime 6.10.2 -- Qt compiled 6.10.2` instead of `PyQt6 6.10.2 ...`). Moreover, according to [PySide6 on PyPI](https://pypi.org/project/PySide6/), "PySide6 is the official Python module from the https://wiki.qt.io/Qt_for_Python", so it seems usual for users to have PySide6 installed.

Last but not least, when I ignored the failing tests and proceeded with the upgrade, `apol` launched without any obvious issue.
The `TypeError` exception seems to only be raised when executing tests (and in the test report, statement `raise TypeError(...)` is located in module `pytestqt`). So this is really about failing tests, even tough the real GUI program seems to run fine (from a quick test).

Would you know what could cause this compatibility issue of SETools' GUI test suite with PySide6?

If you need more context to reproduce this issue, I wrote a comment in the downstream issue: https://github.com/archlinuxhardened/selinux/issues/141#issuecomment-3864470956
I am using a x86_64 computer and the package versions related to Python and Qt are:

```console
$ pacman -Q cython pyside6 python python-networkx python-pyqt6 python-pytest python-pytest-qt python-setuptools \
  qt6-base qt6-declarative qt6-translations
cython 3.2.4-3
pyside6 6.10.2-1
python 3.14.2-2
python-networkx 3.5-2
python-pyqt6 6.10.2-3
python-pytest 1:8.4.2-3
python-pytest-qt 4.4.0-5
python-setuptools 1:80.9.0-4
qt6-base 6.10.2-1
qt6-declarative 6.10.2-1
qt6-translations 6.10.2-1
```

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -26,25 +26,32 @@ jobs:
       fail-fast: false
 
       matrix:
-        build-opts:
-          - {python: '3.10', tox: python3}
-          - {python: '3.11', tox: python3}
-          - {python: '3.12', tox: python3}
-          - {python: '3.13', tox: python3}
+        # unit tests
+        python:
+          - "3.10"
+          - "3.11"
+          - "3.12"
+          - "3.13"
+          - "3.14"
+        tox:
+          - "python3"
+        include:
+          # static analysis
           - {python: '3.11', tox: pep8}
           - {python: '3.11', tox: lint}
           - {python: '3.11', tox: mypy}
           - {python: '3.11', tox: coverage}
+          # install test
           - {python: '3.11', tox: install}
 
     steps:
     - uses: actions/checkout@v3
 
     # This should be the minimum required Python version to build refpolicy.
-    - name: Set up Python ${{ matrix.build-opts.python }}
+    - name: Set up Python ${{ matrix.python }}
       uses: actions/setup-python@v6
       with:
-        python-version: ${{ matrix.build-opts.python }}
+        python-version: ${{ matrix.python }}
 
     - name: Install dependencies
       run: |
@@ -90,6 +97,6 @@ jobs:
       run: |
         # confirm python version
         python --version
-        python -m tox -e ${{ matrix.build-opts.tox }}
+        python -m tox -e ${{ matrix.tox }}
       env:
         LD_LIBRARY_PATH: "${{env.SEPOL_SRC}}/src:${{env.SELINUX_SRC}}/src:${LD_LIBRARY_PATH}"

--- a/README.md
+++ b/README.md
@@ -76,17 +76,6 @@ and perform the following at the root:
 This will compile the C portion of SETools locally, and then
 the tools can be ran from the current directory (e.g. `./seinfo`).
 
-### Rebuilding the Apol Help File
-
-For convenience, a prebuilt copy of the apol help data file is included.
-To rebuild this file, the Qt5 development tools are required
-(particularly, the `qcollectiongenerator` tool).  At the root
-of the SETools sources, perform the following:
-
-```bash
-  $ python setup.py build_qhc
-```
-
 ### Installing SETools
 
 Unpack the official distribution or check out the git repository,

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -199,10 +199,6 @@ persistent = true
 # Discover python modules and packages in the file system subtree.
 # recursive =
 
-# When enabled, pylint would attempt to guess common misconfiguration and emit
-# user-friendly hints instead of false-positive error messages.
-suggestion-mode = true
-
 # Allow loading of arbitrary C extensions. Extensions are imported into the
 # active Python interpreter and may run arbitrary code.
 # unsafe-load-any-extension =

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -646,6 +646,7 @@ redefining-builtins-modules = ["six.moves", "past.builtins", "future.builtins", 
 [tool.pytest.ini_options]
 addopts = ["--import-mode=importlib",]
 pythonpath = "."
+qt_api = "pyqt6"
 markers = [
     "obj_args"
 ]

